e2e/app/component/concept-domain-card.ts

```diff
@@ -0,0 +1,68 @@
+import {Page} from 'puppeteer';
+import {waitWhileLoading} from 'utils/test-utils';
+import Button from 'app/element/button';
+import Container from 'app/container';
+
+export enum Domain {
+   Conditions = 'Conditions',
+   DrugExposures = 'Drug Exposures',
+   Measurements = 'Measurements',
+   Observations =  'Observations',
+   Procedures = 'Procedures',
+}
+
+const domainCardSelector = {
+  cardXpath: '//*[@data-test-id="domain-box"]',
+}
+
+export default class ConceptDomainCard extends Container {
+
+  static findDomainCard(page: Page, domain: Domain): ConceptDomainCard {
+    const selector = `${domainCardSelector.cardXpath}[child::*[@tabindex="0" and @role="button" and text()="${domain}"]]`;
+    return new ConceptDomainCard(page, selector);
+  }
+
+  private constructor(page: Page, xpath: string) {
+    super(page, xpath);
+  }
+
+  async clickSelectConceptButton(): Promise<void> {
+    const selectConceptButton = await this.getSelectConceptButton();
+    await selectConceptButton.click();
+    await this.page.waitForXPath('//*[@data-test-id="conceptTable"]', {visible: true});
+    await waitWhileLoading(this.page);
+  }
+
+   /**
+    * Find displayed number of Participants in this domain.
+    */
+  async getParticipantsCount(): Promise<string> {
```

Comment:
these 2 functions are nearly identical, so we can use common code

Reply:
yes. made a common function
```
private async extractConceptsCount(selector: string): Promise<string> {
    const elemt = await this.page.waitForXPath(selector, {visible: true});
    const textContent = await elemt.getProperty('textContent');
    const value = (await textContent.jsonValue()).toString();
    const regex = new RegExp(/\d{1,3}(,?\d{3})*/); // Match numbers with comma
    return (regex.exec(value))[0];
  }
```
new function is called in `getConceptsCount()` and `getParticipantsCount()`.